lib/logical_solver/presolve.hpp

```diff
@@ -0,0 +1,455 @@
+#include <string>
+#include <vector>
+#include <stack>
+#include <limits>
+#include <cmath>
+#include <iomanip>
+
+namespace logical_solver{
+
+  class Presolve
+  {
+  public:
+  // PUBLIC INSTANCE VARIABLES:
+  // problem
+  std::vector<std::vector<int>> problem_matrix_;
+  std::vector<double> lower_bounds_;
+  std::vector<double> upper_bounds_;
+
```

Comment:
most of this should probably be private (think what you should be able to mess with from outside the class methods which is basically none of it).

Reply:
Done.